ui/Main.elm

```diff
@@ -0,0 +1,341 @@
+viewSystemAddonVersionsRow addon =
+    Html.tr []
+        [ Html.td [] [ Html.text addon.id ]
+        , Html.td [] [ Html.text <| Maybe.withDefault "" addon.builtin ]
+        , Html.td [] [ Html.text <| Maybe.withDefault "" addon.update ]
+        ]
+
+
+viewSystemAddons : List SystemAddon -> Maybe (List SystemAddon) -> Html.Html Msg
+viewSystemAddons builtins updates =
+    Html.table []
+        [ Html.thead []
+            [ Html.td [] [ Html.text "Id" ]
+            , Html.td [] [ Html.text "Built-in" ]
+            , Html.td [] [ Html.text "Updated" ]
+            ]
+        , Html.tbody [] <|
+            List.map viewSystemAddonVersionsRow <|
+                joinBuiltinsUpdates builtins updates
+        ]
+
+
+viewRelease : Release -> Html.Html Msg
+viewRelease { details, builtins, updates } =
+    Html.div []
+        [ Html.h2 [] [ Html.text details.filename ]
+        , viewReleaseDetails details
+        , Html.dl []
+            [ Html.dt [] [ Html.text "System Addons" ]
+            , Html.dd [] [ viewSystemAddons builtins updates ]
+            ]
+        ]
+
+
+viewFilters : Model -> Html.Html Msg
+viewFilters model =
+    let
+        channelView channel =
+            Html.li []
+                [ Html.label []
+                    [ Html.input
+                        [ Html.Attributes.type_ "checkbox"
+                        , Html.Attributes.value channel
+                        , Html.Events.onClick <| ToggleChannelFilter channel
+                        , Html.Attributes.checked <| hasChannelFilter model channel
+                        ]
+                        []
+                    , Html.text channel
+                    ]
+                ]
+    in
+        Html.div []
+            [ Html.h2 [] [ Html.text "Filters" ]
+            , Html.h3 [] [ Html.text "Channels" ]
+            , Html.ul [] <| List.map channelView model.channels
+            ]
+
+
+view : Model -> Html.Html Msg
+view model =
+    Html.div []
+        [ viewFilters model
+        , Html.div [] <| List.map viewRelease model.filteredReleases
+        ]
+
+
+extractChannels : List Release -> List Channel
+extractChannels releaseList =
+    List.map (.details >> .channel) releaseList
+        |> Set.fromList
+        |> Set.toList
+
+
+hasChannelFilter : Model -> Channel -> Bool
+hasChannelFilter model channel =
+    List.any (\x -> x == channel) model.filters.channels
+
+
+filterReleaseChannels : Model -> Channel -> Model
+filterReleaseChannels model channel =
+    let
+        channels =
+            if hasChannelFilter model channel then
+                List.filter (\x -> x /= channel) model.filters.channels
+            else
+                channel :: model.filters.channels
+    in
+        { model
+            | filters = { channels = channels }
+            , filteredReleases =
+                List.filter
+                    (\r -> List.member r.details.channel channels)
+                    model.releases
```

Comment:
Is it really necessary to store this as an attribute of the model, vs. a function that takes `Model` as an argument?